server/src/main/java/org/elasticsearch/common/lucene/Lucene.java

```diff
@@ -1085,4 +1086,33 @@ public static void scanSeqNosInReader(DirectoryReader directoryReader, long from
             }
         }
     }
+
+    /**
+     * Returns a new Bits which is the union of the given Bits and a DocIdSetIterator.
+     * If the DocIdSetIterator is empty, the same Bits instance will be returned.
+     */
+    public static Bits union(Bits bits, DocIdSetIterator iterator) throws IOException {
```

Comment:
we can also use `Lucene#asSequentialAccessBits` instead and then just intersect the two bitsets?